src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -3194,7 +3194,19 @@ private void BuildLiteralProperties(IIndentedStringBuilder writer, XamlObjectDef
 								var isInitializableCollection = IsInitializableCollection(member.Member);
 								var isInlineCollection = IsInlineCollection(member.Member, nonBindingObjects);
 
-								if (isInitializableCollection || isInlineCollection)
+								if (isInlineCollection && closureName != null)
```

Comment:
`closureName != null` indicates that it's a top-level element? Comments would help here

Reply:
It's more when it's in a closure we can't use the "inline" initialization, it's just not something we can do in C#.

It happends we're in a closure for the root properties, but the bug is not limited to root properties.